src/Features/Core/Portable/ChangeAccessibilityModifier/AbstractChangeAccessibilityModifierCodeFixProvider.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.ChangeAccessibilityModifier
+{
+    internal abstract class AbstractChangeAccessibilityModifierCodeFixProvider : CodeFixProvider
+    {
+        private readonly string _title = FeaturesResources.Change_accessibility;
+        private readonly string _titleFormat = FeaturesResources.Change_accessibility_to_0;
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var cancellationToken = context.CancellationToken;
+            var document = context.Document;
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            // Innermost: We are looking for an IdentifierName. IdentifierName is sometimes at the same span as its parent (e.g. SimpleBaseTypeSyntax).
+            var diagnosticNode = root.FindNode(context.Span, getInnermostNodeForTie: true);
+            if (!syntaxFacts.IsDeclaration(diagnosticNode))
+            {
+                return;
+            }
+
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var declaredSymbol = semanticModel.GetDeclaredSymbol(diagnosticNode, cancellationToken);
+            if (declaredSymbol is not (IPropertySymbol or IMethodSymbol or IEventSymbol))
+            {
+                return;
+            }
+
+            var diagnostic = context.Diagnostics[0];
+
+            if (declaredSymbol.IsOverride)
+            {
+                // only show accessibility of base definition for override
+                var original = declaredSymbol.GetOverriddenMember();
+                if (original is null)
+                {
+                    return;
+                }
+
+                var accessibility = original.ComputeResultantAccessibility(declaredSymbol.ContainingType);
+
+                if (accessibility == Accessibility.Internal
+                    && !original.ContainingAssembly.GivesAccessTo(declaredSymbol.ContainingAssembly))
+                {
+                    // not able to override inaccessible member - return
+                    return;
+                }
+
+                if (accessibility == Accessibility.Private)
+                {
+                    // should be unreachable - return for robustness
+                    return;
```

Comment:
If this should never be reached, I think we can `throw ExceptionUtilities.Unreachable;`.

Reply:
This is in features layer, not compiler layer. For robustness I don't want to throw; just do nothing is OK.

Reply:
@huoyaoyuan I believe ExceptionUtilities is available for use in Features layer through an InternalsVisibleTo.

It's used for example in:

https://github.com/dotnet/roslyn/blob/cf55f3a58e47298426fa971d3bd9d8857c746c65/src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.ActiveFileState.cs#L59

Throwing may reveal wrong assumptions (if any). So IMO it's better than just returning here. But I'd defer to @CyrusNajmabadi.